### PR TITLE
make nomkl the default

### PR DESCRIPTION
**Pull request by bsipocz**

### CODE REVIEW

**Comment by bsipocz** [comment]
(this still needs tests and docs)

**Comment by bsipocz** [comment]
(this still needs tests and docs, but I wanted it out of the way)

File: travis/setup_dependencies_common.sh [hunk 1]
```diff
@@ -221,23 +225,29 @@ fi
 # NUMPY
 # We use --no-pin to avoid installing other dependencies just yet.
 
+
+MKL = 'nomkl'
```
Comment (astrofrog) [inline travis/setup_dependencies_common.sh]: remove spaces?

File: travis/setup_dependencies_common.sh [hunk 2]
```diff
@@ -221,23 +225,29 @@ fi
...
+if [[ ! -z $(echo $CONDA_DEPENDENCIES | grep '\bmkl\b') ]]; then
+    MKL = ''
```
Comment (astrofrog) [inline travis/setup_dependencies_common.sh]: remove spaces?

File: travis/setup_dependencies_common.sh [hunk 3]
```diff
@@ -221,23 +225,29 @@ fi
 # NUMPY
 # We use --no-pin to avoid installing other dependencies just yet.
 
+
+MKL = 'nomkl'
+if [[ ! -z $(echo $CONDA_DEPENDENCIES | grep '\bmkl\b') ]]; then
```
Comment (astrofrog) [inline travis/setup_dependencies_common.sh]: Could you write ``if [[ $CONDA_DEPENDENCIES == *mkl* ]]``? (untested)

File: travis/setup_dependencies_common.sh [hunk 4]
```diff
@@ -221,23 +225,29 @@ fi
...
+    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION $MKL"
 elif [[ $NUMPY_VERSION == stable ]]; then
-    conda install $QUIET --no-pin numpy=$LATEST_NUMPY_STABLE
+    conda install $QUIET --no-pin numpy=$LATEST_NUMPY_STABLE $MKL
     export NUMPY_OPTION="numpy=$LATEST_NUMPY_STABLE"
-    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION numpy=$LATEST_NUMPY_STABLE"
+    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION numpy=$LATEST_NUMPY_STABLE $MKL"
 elif [[ $NUMPY_VERSION == pre* ]]; then
-    conda install $QUIET --no-pin nomkl numpy
+    conda install $QUIET --no-pin $MKL numpy
```
Comment (astrofrog) [inline travis/setup_dependencies_common.sh]: So in the case of Numpy dev and prerelease, we actually want to always force nomkl right?

**Comment by bsipocz** [comment]
It's a tiny bit weird that on travis, on the current master (before this PR), mkl is not installed at all but it is for the astropy travis builds. No I don't have good ideas how to test this PR in this repo (but will test the this branch with the astropy travis).

File: travis/setup_dependencies_common.sh (hunk 3 above)
Comment (astrofrog) [reply travis/setup_dependencies_common.sh]: 👍

File: travis/setup_dependencies_common.sh (hunk 4 above)
Comment (astrofrog) [reply travis/setup_dependencies_common.sh]: 👍

**Comment by bsipocz** [comment]
I think I've addressed all the point of the review, and sadly cannot implement this feature for appveyor as  there is no openblas on Windows, everything needs to ship with mkl (https://github.com/conda/conda/issues/6071#issuecomment-421010440)

**Pull request merged**